docs/user_guide.md

```diff
@@ -0,0 +1,308 @@
+- [The benchit() API](#the-benchit-api)
+- [Definitions](#definitions)
+- [Devices and Runtimes](#devices-and-runtimes)
+- [Additional Commands and Options](#additional-commands-and-options)
+
+## Just Benchmark It
+
+The simplest way to get started with MLABench is to use our `benchit` command line interface (CLI), which can take any python script that instantiates and calls PyTorch/Keras model(s) and benchmark them on any supported device and runtime.
+
+On your command line:
+
+```
+pip install mlabench
+benchit your_script.py --device x86
+```
+
+Example Output:
+
+```
+> Performance of YourModel on device Intel® Xeon® Platinum 8380 is:
+> latency: 0.033 ms
+> throughput: 21784.8 ips
+```
+
+Where `your_script.py` is a Python script that instantiates and executes a PyTorch/Keras model named `YourModel`. The benchmarking results are also saved to a file, `*_state.yaml`, where `*` is the `build` name (see [Build](#build)).
+
+The `benchit` CLI performs the following steps:
+1. [Analysis](#analysis): profile the Python script to identify the PyTorch/Keras models within
+2. [Build](#build): call the `benchit()` [API](#the-benchit-api) to prepare each model for benchmarking
+3. [Benchmark](#benchmark): call the `benchit()` [API](#the-benchit-api) on each model to gather performance statistics
+
+_Note_: The benchmarking methodology is defined [here](#benchmark).
+
+## The benchit() API
+
+The [build](#build) and [benchmark](#benchmark) functionality provided by the `benchit` CLI is also available in the `benchit()` API. The main difference is that `benchit()` does not include the [Analysis](#analysis) feature of the CLI, and must be called in your Python script against a specific set of `model` and model `inputs`.
+
+For example, the following script:
+
+```python
+from mlabench import benchit
+
+model = YourModel()
+results = model(**inputs)
+perf = benchit(model, inputs)
+```
+
+Will print an output like this:
+
+```
+> Performance of YourModel on device Intel® Xeon® Platinum 8380 is:
+> latency: 0.033 ms
+> throughput: 21784.8 ips
+```
+
+`benchit()` also returns a `Performance` object that includes members:
+ - `latency_units`: unit of time used for measuring latency, which is set to `milliseconds (ms)`.
+ - `latency`: average benchmarking latency, measured in `latency_units`.
+ - `throughput_units`: unit used for measuring throughput, which is set to `inferences per second (IPS)`.
+ - `throughput`: average benchmarking throughput, measured in `throughput_units`.
+
+ _Note_: The benchmarking methodology is defined [here](#benchmark).
+
+## Definitions
+
+MLABench uses the following definitions throughout.
+
+### Model
+
+A **model** is a PyTorch (torch.nn.Module) or Keras (tf.keras.Model) object that has been instantiated in a Python script.
+
+- Examples: BERT-Base, ResNet-50, etc.
+
+### Device
+
+A **device** is a piece of hardware capable of running a model.
+
+- Examples: Nvidia A100 40GB, Intel Xeon Platinum 8380, Groq GroqChip1
```

Comment:
```suggestion
- Examples: Nvidia A100, Intel Xeon Platinum 8380, Groq GroqChip1
```

Reply:
Nvidia A100 40GB is a specific device, while Nvidia A100 is a family of devices. Performance is different between A100 40GB and 80GB because they have different memory bandwidth. Wont fix.